metric-example/query-by-example.md

```diff
@@ -0,0 +1,516 @@
+```
+"Host 14.A ~"Alice~" ~~80° (Europe, North Africa)"
+```
+Alternatively, we can rely entirely on escaping instead of quoting, but this gets messy real quick:
+```
+Host~ 14~.A~ "Alice"~ ~~80° ~(Europe~,~ North~ Africa~)
+```
+To filter for this name, we can use a combination of :names and :filter and use our quoted version for the filter matcher to end up with a valid selector that uses special characters:
+```
+builtin:host.disk.avail  
+: names  
+: filter(  
+	eq(  
+		dt.entity.host.name,  
+		"Host 14.A ~"Alice~" ~~80° (Europe, North Africa)"   
+	)  
+ )
+```
+
+# Scenario 16: Two Metrics with Two Different Entity Selectors
+
+>**Task**
+>We want to query multiple metrics at once, and each metric should use a different entity selector.
+
+There is only one `entitySelector` GET parameter, and an entity selector queries for exactly one entity type, but the metric selector can query more than one metric and those metrics could use different entity types.
+
+How can we query the PGI-based metric `builtin:tech.generic.cpu.usage` together with `builtin:host.cpu.usage` with the hosts and process group instances both filtered using a tag "business-critical"?
+
+This is easily possible by embedding entity selectors into filters using the `in` matcher with the right-hand side of `in` supplied using the `entitySelector` function that sets off an entity query from within a metric selector. Since entity selectors use a lot of special characters, the argument to the `entitySelector` function should always be quoted:
+```
+builtin:tech.generic.cpu.usage
+: filter(  
+    in(
+      dt.entity.process_group_instance,  
+      entitySelector("type(PROCESS_GROUP_INSTANCE),tag(business-critical)") 
+  	)  
+),
+
+builtin:host.cpu.usage  
+: filter(  
+     in(  
+       dt.entity.host,  
+       entitySelector("type(HOST),tag(business-critical")  
+     )  
+)
+```
+
+# Scenario 17: Does our Service use Less Memory after the Update
+
+>**Task**
+>We want to see the CPU utilization graph of yesterday and the day before yesterday, displayed on top of each other in a chart, so that we can look out for differences.
+
+A specific process group instance has been running on a new version yesterday. We want to display a graph of yesterdays CPU usage and for comparison, we want the day before layered on top of yesterday. We get yesterday with `from=now/d-d&to=now/d`. This time window can then be moved per each queried metric by using the `timeshift` operator, which takes a time interval to move the data by, but keeping the original timestamps, so that it can be layered on top of the other graph. In our case, with the time frame from before we need `timeshift(-1d)`:

```

Comment:
```suggestion
A specific process group instance ran on a new version yesterday. We want to display a graph of yesterday's CPU usage and for comparison, we want the day before layered on top of yesterday. We get yesterday with `from=now/d-d&to=now/d`. This time window can then be moved per each queried metric by using the `timeshift` operator, which takes a time interval to move the data by, but keeping the original timestamps, so that it can be layered on top of the other graph. In our case, with the timeframe from before we need `timeshift(-1d)`:

```